cmd/orphan-finder/main.go

```diff
@@ -0,0 +1,190 @@
+package main
+
+import (
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/cactus/go-statsd-client/statsd"
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/codegangsta/cli"
+
+	"github.com/letsencrypt/boulder/cmd"
+	"github.com/letsencrypt/boulder/core"
+	blog "github.com/letsencrypt/boulder/log"
+	"github.com/letsencrypt/boulder/rpc"
+)
+
+type config struct {
+	AMQP   cmd.AMQPConfig
+	Statsd cmd.StatsdConfig
+	Syslog cmd.SyslogConfig
+}
+
+var (
+	b64derOrphan = regexp.MustCompile(`b64der=\[([a-zA-Z0-9+/]+)\]`)
+	regOrphan    = regexp.MustCompile(`regID=\[(\d+)\]`)
+)
+
+func checkDER(sai core.StorageAuthority, der []byte) error {
+	cert, err := x509.ParseCertificate(der)
+	if err != nil {
+		return fmt.Errorf("Failed to parse DER: %s", err)
+	}
+	_, err = sai.GetCertificate(core.SerialToString(cert.SerialNumber))
+	if err != nil {
+		if _, ok := err.(core.NotFoundError); ok {
+			return nil
+		}
+		return fmt.Errorf("Existing certificate lookup failed: %s", err)
+	}
+	return nil
```

Comment:
Sorry, one last thing: If `GetCertificate` succeeded, we wanted to error out, right?